HIRS_AttestationCAPortal/src/main/java/hirs/attestationca/portal/page/controllers/ValidationReportsPageController.java

```diff
@@ -97,4 +114,84 @@ public void modify(final Criteria criteria) {
 
         return new DataTableResponse<>(records, input);
     }
+
+    /**
+     * This method handles downloading a validation report. The report will contain the
+     * following data:
+     * - Company devices where shipped from
+     * - Contract#
+     * - Report for Date range (default to current date)
+     * -Verified Manufacturer is the Platform Vendor
+     * - Model is the Platform Model
+     * - SN is the Chassis SN
+     * - Verification Data is the not before time on the Attestation Certificate
+     * - Component Status column is 8 component classes names listed above
+     * (Component Status data is taken from the pass/fail status of the report summary)
+     * - Device Status is the overall pass/fail of the report summary
+     * @param id of the validated device
+     * @param deviceName name of the validated device
+     * @param request object
+     * @param response object
+     * @throws IOException thrown by BufferedWriter object
+     */
+    @RequestMapping(value = "download", method = RequestMethod.POST)
+    public void download(@RequestParam final String id,
+                         @RequestParam final String deviceName,
+                         final HttpServletRequest request,
+                         final HttpServletResponse response) throws IOException {
+
+        LOGGER.info("Downloading validation report for " + id);
+        response.setHeader("Content-Type", "text/plain");
+        response.setHeader("Content-Disposition",
+                "attachment;filename=\"" + deviceName + "_validation_report.txt\"");
+        BufferedWriter bufferedWriter = new BufferedWriter(
+                new OutputStreamWriter(response.getOutputStream(), "UTF-8"));
+        Enumeration parameters = request.getParameterNames();
+        while (parameters.hasMoreElements()) {
+            String parameter = (String) parameters.nextElement();
+            bufferedWriter.append(parameter + ": " + request.getParameter(parameter) + "\n");
+            LOGGER.info(parameter + ": " + request.getParameter(parameter));
+        }
+//        String columnHeaders = "Company, Contract Number, Date Range, Verified Manufacturer, "
```

Comment:
commented code, do you mean to leave this here?

Reply:
Yes, these are the column headers in case we keep the csv format